pkg/istio-agent/health/health_probers.go

```diff
@@ -52,19 +53,13 @@ func (p *ProbeResult) IsHealthy() bool {
 	return *p == Healthy
 }
 
-func (p *ProbeResult) IsUnhealthy() bool {
-	return *p == Unhealthy
-}
-
-func (p *ProbeResult) IsUnknown() bool {
-	return *p == Unknown
-}
-
 type HTTPProber struct {
 	Config    *v1alpha3.HTTPHealthCheckConfig
 	Transport *http.Transport
 }
 
+var _ Prober = &HTTPProber{}
```

Comment:
why are we putting ignored probers here?

Reply:
compile-time assertion that `&HTTPProber` implements the `Prober` interface